packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx

```diff
@@ -441,54 +441,50 @@ export class DatePickerBase extends BaseComponent<IDatePickerProps, IDatePickerS
 
     if (allowTextInput) {
       let date = null;
-      if (inputValue) {
-        // Don't parse if the selected date has the same formatted string as what we're about to parse.
-        // The formatted string might be ambiguous (ex: "1/2/3" or "New Year Eve") and the parser might
-        // not be able to come up with the exact same date.
-        if (this.state.selectedDate && formatDate && formatDate(this.state.selectedDate) === inputValue) {
-          return;
-        } else {
-          date = parseDateFromString!(inputValue);
-
-          // Check if date is null, or date is Invalid Date
-          if (!date || isNaN(date.getTime())) {
-            // Reset invalid input field, if formatting is available
-            if (formatDate) {
-              date = this.state.selectedDate;
-              this.setState({
-                formattedDate: formatDate(date!).toString()
-              });
-            }
+      // Don't parse if the selected date has the same formatted string as what we're about to parse.
+      // The formatted string might be ambiguous (ex: "1/2/3" or "New Year Eve") and the parser might
+      // not be able to come up with the exact same date.
+      if (inputValue && !(this.state.selectedDate && formatDate && formatDate(this.state.selectedDate) === inputValue)) {
```

Comment:
Does this bug repro in the @uifabric/date-time DatePicker also? We might need to make the same fix there.